### ISSUE TITLE
NiftiMasker should "time-mask" 4D volumes

### ISSUE THREAD

**Issue by VirgileFritsch** [nilearn/nilearn#165]
In examples (e.g. plot_haxby_searchlight.py), we often do something like:

```python
fmri_img = nibabel.load(dataset_files.func)

### Restrict to faces and houses ################################
condition_mask = np.logical_or(conditions == 'face')
fmri_img = nibabel.Nifti1Image(fmri_img.get_data()[..., condition_mask],
                               fmri_img.get_affine().copy())

### Mask data ###############################################
mask_img = nibabel.load(dataset_files.mask)
nifti_masker = NiftiMasker(mask=mask_img)
fmri_masked = nifti_masker.fit_transform(fmri_img)
```

It would be simpler to provide `condition_mask` directly to the NiftiMasker to avoid creating a temporary image.

On top of making things easier to use, one important consideration is that some preprocessings performed by the NiftiMasker (such as detrending) might need the complete time serie to be accurate.
The NiftiMasker should then of course be able to choose whether it has to process the whole time serie or only the desired volumes according to the preprocessings that have to be performed. In the first case, is it important not to have masked the data before, while computation time would be saved in the second case (potentially a lot).
